runtime/runtime_controller.h

```diff
@@ -26,6 +26,26 @@ class RuntimeDelegate;
 class View;
 class Window;
 
+struct WindowData {
+  WindowData();
+
+  WindowData(const WindowData& other);
```

Comment:
Why do we need a custom copy constructor?  If we need one we should probably do something about operator= as well.